docs/features/module-initializers.md

```diff
@@ -0,0 +1,109 @@
+# Module Initializers
+## Overview
+
+A module initializer is a method that runs eagerly and one-time when an assembly is loaded, similar to a type initializer (static constructor).
```

Comment:
```suggestion
A module initializer is a method that runs before the first access of a field or invocation of a method in that module, similar to a type initializer (static constructor).
```

The difference is that `Assembly.Load` (and even subsequent `Assembly.GetTypes`) do not cause the module initializer to run. I'd describe it as lazy rather than eager.